doc/book/box/data_model.rst

```diff
@@ -554,6 +554,104 @@ Charts explaining the precise differences from DUCET order are
+`referential integrity <https://en.wikipedia.org/wiki/Referential_integrity>`_
+of the database.
+
+Some fields can only contain values present in other spaces. For example,
+shop orders always belong to existing customers. Hence, all values of the ``customer``
+field of the ``orders`` space must exist in the ``customers`` space. In this case,
+``customers`` is a **parent space** for ``orders`` (its **child space**). Foreign keys
+enable automatic checks of values presence in the parent space upon each insertion and
+modification of the child space.
+
+*****************
+Foreign key types
+*****************
+
+There are two types of foreign keys in Tarantool:
+
+* *field foreign keys* check that the value being assigned to a field
+  is present in a particular field of another space. For example, the ``customer``
+  value of an ``orders`` tuple must match an ``id`` stored in the ``customers`` space.
+
+* *tuple foreign keys* check that multiple fields of a tuple have a match in
+  another space. For example, if the ``orders`` space has fields ``customer_id``
+  and ``customer_name``, a tuple foreign key can check that the ``customers`` space
+  contains a tuple with both these values in the corresponding fields.
+
+Field foreign keys work faster while tuple foreign keys allow implementing
+more strict references.
+
+***********************
+Creating foreign keys
+***********************
+
+.. important::
+
+  For all foreign keys, there must exist indexes that include their fields.
+
+To create a foreign key in a space, specify the parent space and linked fields in the ``foreign_key`` parameter.
+Fields can be referenced by name or by number:
+
+* Field foreign keys: when setting up the space format:
+
+  .. code-block:: tarantoolsession
+
+      tarantool> box.space.orders:format({
+               > {name = 'id',   type = 'number'},
+               > {name = 'customer_id', foreign_key = {space = 'customers', field = 'id'}}, -- or field = 1
+               > {name = 'price_total',  type = 'number'},
+               > })
+
+* Tuple foreign keys: when creating or altering a space. Note that for foreign
+  keys with multiple fields there must exist an index that includes all these fields.
+
+  .. code-block:: tarantoolsession
+
+      tarantool> box.schema.space.create("orders", {foreign_key={space='customers', field={customer_id='id', customer_name='name'}}})
+      ---
+      ...
+      tarantool> box.space.orders:format({
+               > {name = "id", type = "number"},
+               > {name = "customer_id" },
+               > {name = "customer_name"},
+               > {name = "price_total",    type = "number"},
+               > })
+
+.. note::
+
+  Type can be omitted for foreign key fields because it's
+  defined in the parent space.
+
+Foreign keys can have an optional name.
+
+.. code-block:: lua
+
+    foreign_key = {customer = {space = '...', field = {...}}}
+
+A space can have multiple named tuple foreign keys:
```

Comment:
This long line of nouns adjunct leaves me puzzled. Consider rephrasing the sentence.